unliftio/test/UnliftIO/PooledAsyncSpec.hs

```diff
@@ -0,0 +1,174 @@
+module UnliftIO.PooledAsyncSpec (spec) where
+
+import Test.Hspec
+import Control.Concurrent
+import Data.List (nub, sort)
+import Test.QuickCheck
+import Data.Functor ((<$>))
+import UnliftIO
+
+spec :: Spec
+spec = do
+  let exAction :: Int -> IO Int
+      exAction x = do
+        if (x == 2) then error "hell" else return ()
+        return x
+               
+      action :: Int -> IO ThreadId
+      action x = do
+         threadDelay (2 * 10^6)
```

Comment:
Do we really need to run this action for 2 seconds? If we delay just for a fraction of a second `threadDelay (2 * 10 ^ 5)`, we reduce time it takes to execute the test suite 3 times. (from 17 sec -> 6 sec on my computer) .

Reply:
Done.